In [2]:
import pandas as pd

import numpy as np

import matplotlib.pyplot as plt

import seaborn as sns

import scipy.stats as stats

import statsmodels.api as sm

%matplotlib inline

In [9]:
gdp_df = pd.read_csv("../data/gdp_per_capita.csv")
gdp_df.head()

Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2018  1734.723214              NaN
1     Afghanistan  2017  1758.465636              NaN
2     Afghanistan  2016  1757.023490              NaN
3     Afghanistan  2015  1766.593077              NaN
4     Afghanistan  2014  1795.735834              NaN

In [16]:
gdp_df = gdp_df.drop('Value Footnotes', axis = 1)

In [24]:
gdp_df.rename(columns = {"Country or Area": "Country"},  
          inplace = True) 

In [25]:
gdp_df.rename(columns = {"Value":"GDP_Per_Capita"},  
          inplace = True) 
gdp_df.head()

Country  Year  GDP_Per_Capita
0  Afghanistan  2018     1734.723214
1  Afghanistan  2017     1758.465636
2  Afghanistan  2016     1757.023490
3  Afghanistan  2015     1766.593077
4  Afghanistan  2014     1795.735834

In [28]:
gdp_df.shape

(6731, 3)

### Question 7

gdp_df has 6,731 rows and 3 columns

Types:
Country: Object
Year: Object
GDP_Per_Capita: Float

In [55]:
gdp_df.dtypes

Country            object
Year                int32
GDP_Per_Capita    float64
dtype: object

In [39]:
gdp_df.dropna(inplace = True)
gdp_df['Year'] = gdp_df['Year'].astype(int)

In [40]:
gdp_df.dtypes

Country            object
Year                int32
GDP_Per_Capita    float64
dtype: object

In [43]:
print(gdp_df['Year'].min())
print(gdp_df['Year'].max())

1990
2018


In [47]:
gdp_df.groupby(by=['Year']).sum()

GDP_Per_Capita
Year                
1990    2.456235e+06
1991    2.464479e+06
1992    2.486493e+06
1993    2.508508e+06
1994    2.559041e+06
1995    2.730769e+06
1996    2.789179e+06
1997    2.944708e+06
1998    3.076481e+06
1999    3.149284e+06
2000    3.440492e+06
2001    3.481548e+06
2002    3.524402e+06
2003    3.602727e+06
2004    3.755947e+06
2005    3.859435e+06
2006    4.010457e+06
2007    4.154704e+06
2008    4.192940e+06
2009    4.059157e+06
2010    4.148641e+06
2011    4.247377e+06
2012    4.287388e+06
2013    4.332068e+06
2014    4.328233e+06
2015    4.360375e+06
2016    4.408823e+06
2017    4.481321e+06
2018    4.243078e+06

### Question 8
The years represented are 1990 to 2018.
I notice that the GDp increases as the Years get closer and closer to the present. They have a strong positive correlation.

In [54]:
gdp_df.pivot_table(index=['Country'], aggfunc='size')

Country
Afghanistan            17
Albania                29
Algeria                29
Angola                 29
Antigua and Barbuda    29
                       ..
West Bank and Gaza     25
World                  29
Yemen                  29
Zambia                 29
Zimbabwe               29
Length: 243, dtype: int64

### Question 9

How many countries are represented in this dataset? 

Which countries are least represented in the dataset? 

Why do you think these countries have so few observations?